teos/src/dbm.rs

```diff
@@ -1175,7 +1189,61 @@ mod tests {
         dbm.store_tracker(uuid, &tracker).unwrap();
 
         // We should get all the appointments matching our locator back except from the triggered one
-        assert_eq!(dbm.load_appointments(Some(locator)), appointments);
+        assert_eq!(dbm.load_appointments(Some((locator, None))), appointments);
+    }
+
+    #[test]
+    fn test_load_appointments_with_locator_and_user_id() {
+        let dbm = DBM::in_memory().unwrap();
+
+        let mut appointments = HashMap::new();
+        let dispute_tx = get_random_tx();
+        let dispute_txid = dispute_tx.txid();
+        let locator = Locator::new(dispute_txid);
+
+        // create user id
+        let user_id = get_random_user_id();
+        let user = UserInfo::new(AVAILABLE_SLOTS, SUBSCRIPTION_START, SUBSCRIPTION_EXPIRY);
+        dbm.store_user(user_id, &user).unwrap();
+
+        // Create and store a particular appointment
+        let (uuid, appointment) =
+            generate_dummy_appointment_with_user(user_id, Some(&dispute_txid));
+        dbm.store_appointment(uuid, &appointment).unwrap();
+        appointments.insert(uuid, appointment.clone());
+
+        // create random appointments
+        for _ in 1..11 {
+            let (uuid, appointment) = generate_dummy_appointment_with_user(user_id, None);
+            dbm.store_appointment(uuid, &appointment).unwrap();
+            appointments.insert(uuid, appointment);
+        }
+
+        // Returns empty if no appointment matches both userid and locator
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(get_random_user_id()))),),
+            HashMap::new()
+        );
+        assert_eq!(
+            dbm.load_appointments(Some((get_random_locator(), Some(user_id))),),
+            HashMap::new()
+        );
+
+        // Returns particular appointments if they match both userid and locator
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(user_id))),),
+            HashMap::from([(uuid, appointment)])
+        );
+
+        // Create a tracker from existing appointment
+        let tracker = get_random_tracker(user_id, ConfirmationStatus::InMempoolSince(100));
+        dbm.store_tracker(uuid, &tracker).unwrap();
+
+        // ensure that no tracker is returned
```

Comment:
Actually, here you are checking that no *appointment* is returned. This is because there is a tracker with the same uuid, so the appointment is seen as already triggered.

```suggestion
        // Check how, if the appointment is triggered (there's a tracker for it) we don't get an appointment back
```